### ISSUE TITLE
[bitnami/janusgraph] Helm Pod start in failure caused by secret not found

### ISSUE THREAD

**Issue by olivierlemee** [bitnami/charts#26354]
### Name and Version

bitnami/janusgraph 0.3.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. in Minikube 1.33.1 with Kubernetes 1.30.0 kubectl on MacOs Monterey version 12.7.5
2. After added Bitnami helm repository
3. from command line, execute command "helm instal knr-system oci://registry-1.docker.io/bitnamicharts/janusgraph" as written in project documentation for installation ont the Minikube cluster already started and in stable status
4. I see that: knr-system-cassandra-0 Pod is started and running with success
5. but the knr-system-janusgraph-xxxxxxxx Pod stay in "CreateContainerConfig" status with a logs that say ".... is waiting to start: PodInitializing"
6. on the janusgraph Pod in failure, I see a permanent event that give failure cause equals to "knr-system-janusgraph-cassandra" not found message
7. another type of event that appear for "Pulled" message cause regarding the Sub-object "spec.initContainers(wait-for-storage)" is shown confirming that "Container image "docker.io/bitnami/janusgraph:1.0.0-debian-12-r4" already present on machine"
8. the system stay permanently in this unstable status without deployment object (named knr-system-janusgraph) in success
9. and the replicat sets "knr-system-janusgraph-xxxx" blocke on status equals to "Init:CreateContainerConfig" regarding not found secret "knr-system-janusgraph-cassandra"

### Are you using any custom parameters or values?

Yes only a defined nodeSelector defined into the Cassandra and Janusgraph values.yaml files (I downloaded the origin Helm project source codes for try a Helm full installation since original code... but without success start of Pods because have the same error)

### What is the expected behavior?

Start with success of each PODS (janusgraph, cassandra) and finalized deployment with success into a Minikube cluster ready for use

### What do you see instead?

- In janusgraph Pod console, event is show regarding cause of failure equals to:
Error: secret "knr-system-janusgraph-cassandra" not found
- In Minikube Secret object, I see that a secret named "knr-system-cassandra" is existing, but that have not the same name of the Pod failure cause which talk about "knr-system-janusgraph-cassandra"

### Additional information

Perhaps that the different name of searched secret attempted to be read ("knr-system-cassandra") by the Janusgraph Pod in place of the "knr-system-janusgraph-cassandra" defined in the secrets is a problem.
I am blocked since 2 days on this problem to have a success deployed Helm project but I am not find the root cause.
Thanks for your help. Regards

**Comment by olivierlemee** [comment]
I found a solution that allow to go to a next step : 

1. I set the nameOverride property ito the values.yaml of janusgraph helm project
2. I start the install via helm command line
3. I see that error regarding not found secret does not appear because now, it's the equals name "knr-system-cassandra" which is defined into the Pod resource defined into the initialized janusgraph Pod (which is not found regarding the existing secret object instantiated into the Minikube Secrets)
4. BUT... in the Pod events log, I see the after the "Created" log (message = Created container wait-for-storage) regarding the sub-object spec.initContainers(wait-for-storage), there is a log for "Pulled" cause and the Pod stay permanently into a "Init: 0/1" status (I see conditions where "containers with incomplete status: [wait-for-storage]" like causes for ContainersReady and Read and Initialized types which are in a "false" state
5. So, Pod stay in unready status...

**Comment by andresbono** [comment]
Hi, thank you for reporting this issue. I have just sent this PR #27074 that should fix the problems with the secret name.

**Comment by olivierlemee** [comment]
When I try to search in resources started by the helmified project, I see that the ConfigMap instance created for the Janusgraph pod (and generated by the helm project) is:

```
{
	"janusgraph.properties": "storage.backend=cql
		storage.hostname=my-release-cassandra
		storage.port=9043
		cache.db-cache=true"
}
```

but the statefull state that have been by default created is named [my-release-knr-system-cassandra]

So is it normal value created into the janusgraph properties because I think that could be an issue during the Janusgraph pod initialization which does not find the storage.hostname using a unexisting real name deployed in the cluster.

A potential way of issue resolution regarding the pod blocked status for cause of "waiting-for-storage" ?

**Comment by olivierlemee** [comment]
@andresbono Hello, I am sorry but I just try to execute the new version 0.3.8 and I see that failure this time is also on cassandra pods. So the situation is worse than before because the cassandra pods is not not starting with success for cause of "CrashLoopBackOff
Message
back-off 1m20s restarting failed container=cassandra pod=my-release-knr-system-cassandra-0_default(cdd4f1e2-7bcb-43e9-a742-eb6eb7d4c8cb)"
So the ticket have been closed, but the issue is always existing

**Comment by andresbono** [comment]
Hi @olivierlemee, let's recap.

The initially reported issue "Error: secret [...] not found" has **not** been completely resolved by #27074. There are still issues when installing `0.3.8` with the following values (extracted from your comment https://github.com/bitnami/charts/issues/26354#issuecomment-2136809646):

```yaml
cassandra:
  nameOverride: "knr-system-cassandra"
nameOverride: "knr-system"
```

I have submitted a new PR: #27151 that I hope it resolves the remaining issues:
- "Error: secret [...] not found"
- pod blocked status for cause of "waiting-for-storage"

---

On the other hand, you are now additionally facing issues with the cassandra pods. I think this started happening when you installed `0.3.8` but it was fine in `0.3.7`? Please, confirm. There is no change in the cassandra subchart between those 2 versions, so it is strange.

You will need to check the pod logs to know more about this new issue.

**Comment by olivierlemee** [comment]
Hello, I am sorry but I does not understand your comment. Currently I try to use the default cassandra pod initialized by the janusgraph Helm project (that is include a dependency to cassandra image also supported by your sub-graph folder).

Today I try again to deploy the default project from the new available --version 0.3.12, into a specific new created cluster based on unique Node based on more resource (more storage space, more cpus), and it's a success finalized deployment:

- cluster creation command line:
  `minikube start --driver=hyperkit --container-runtime=docker --profile dev --nodes 1 --cpus 12 --disk-size '100g' --memory '64g'`

I think that error and unstable status that was in failure, have been caused by missing resources allocation on the node... but no where this information can be detected.

### HELM PROJECT IMPROVEMENT PROPOSAL
Perhaps could you add a **detailed documentation into the helm project regarding the minimum requirements** (hardware, vCPUS, RAM, storage) that are required for the a success use of your helm project in terms à "minimum requirements" for helping the users to avoid to arrive into the unstable situation where I was.

Thanks for your assistance and issue resolution help.

I hope that your helm project will evolute to allow us (software editor interested to build new application software supported by Janusgraph, ready to be a sub-layer of AI capabilities).

Best regards

**Comment by andresbono** [comment]
Thank you for your suggestion, @olivierlemee. Please note that we configure resource requests and limits as explained at [Resource requests and limits](https://github.com/bitnami/charts/tree/main/bitnami/cassandra#resource-requests-and-limits):

https://github.com/bitnami/charts/blob/055dab873a739e58cba8ffdc4a3d6ec5523e2df0/bitnami/cassandra/values.yaml#L337

https://github.com/bitnami/charts/blob/055dab873a739e58cba8ffdc4a3d6ec5523e2df0/bitnami/common/templates/_resources.tpl#L32-L35

That covers the specification of hardware requirements for the charts.

Also, before releasing new versions, we test the charts in a wide variety of k8s clusters to make sure that those presets are valid.

Maybe by using minikube gives you more instability in your particular scenario.
